server.c

```diff
@@ -0,0 +1,233 @@
+    else if (typeflag == FTW_DNR)
+        printf(" %s/ (unreadable)\n", filepath);
+    else
+        printf(" %s (unknown)\n", filepath);
+
+    return 0;
+}
+
+int print_directory_tree(const char *const dirpath)
+{
+    int result;
+
+    /* Invalid directory path? */
+    if (dirpath == NULL || *dirpath == '\0')
+        return errno = EINVAL;
+
+    result = nftw(dirpath, print_entry, USE_FDS, FTW_PHYS);
+    if (result >= 0)
+        errno = result;
+
+    return errno;
+}
+
+// Function designed for chat between client and server.
+void func(int sockfd)
+{
+    char buff[MAX];
+    int n;
+    // infinite loop for chat
+    for (;;)
+    {
+        memset(buff, 0, MAX);
+
+        // read the message from client and copy it in buffer
+        read(sockfd, buff, sizeof(buff));
+        // print buffer which contains the client contents
+        printf("From client: %s\t To client : ", buff);
+
+        if (strncmp("files", buff, 5) == 0)
+        {
+            printf("List of files");
+            print_directory_tree(".");
+        }
+
+        memset(buff, 0, MAX);
+        n = 0;
+        // copy server message in the buffer
+        while ((buff[n++] = getchar()) != '\n')
+            ;
+
+        // and send that buffer to client
+        write(sockfd, buff, sizeof(buff));
+
+        // if msg contains "Exit" then server exit and chat ended.
+        if (strncmp("exit", buff, 4) == 0)
+        {
+            printf("Server Exit...\n");
+            break;
+        }
+    }
+}
+
+// Driver function
+int main()
+{
+    int sockfd, connfd, len;
+    struct sockaddr_in servaddr, cli;
+
+    // socket create and verification
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    if (sockfd == -1)
+    {
+        printf("socket creation failed...\n");
+        exit(0);
+    }
+    else
+        printf("Socket successfully created..\n");
+    memset(&servaddr, 0, sizeof(servaddr));
+
+    // assign IP, PORT
+    servaddr.sin_family = AF_INET;
+    servaddr.sin_addr.s_addr = htonl(INADDR_ANY);
+    servaddr.sin_port = htons(PORT);
+
+    // Binding newly created socket to given IP and verification
+    if ((bind(sockfd, (SA *)&servaddr, sizeof(servaddr))) != 0)
+    {
+        printf("socket bind failed...\n");
+        exit(0);
+    }
+    else
+        printf("Socket successfully binded..\n");
+
+    // Now server is ready to listen and verification
+    if ((listen(sockfd, 5)) != 0)
+    {
+        printf("Listen failed...\n");
+        exit(0);
+    }
+    else
+        printf("Server listening..\n");
+    len = sizeof(cli);
+
+    // Accept the data packet from client and verification
+    connfd = accept(sockfd, (SA *)&cli, &len);
+    if (connfd < 0)
+    {
+        printf("server acccept failed...\n");
+        exit(0);
+    }
+    else
+        printf("server acccept the client...\n");
+
+    // Function for chatting between client and server
+    func(connfd);
+
+    // After chatting close the socket
+    close(sockfd);
+}
```

Comment:
int main function is not returning any status. Return 0 for success or -1 for error.